### DISCUSSION TITLE
Feature request: send url to device.

### DISCUSSION THREAD

**Discussion by xusiai** [andyholmes/valent#574] [Feature Request]
With the KDE Connect Android application it is possible to share URLs to a device (e.g. open on desktop). Would it be possible to implement this function for the opposite direction e.g. open <url> on phone from the desktop client of Valent?

**Comment by andyholmes** [comment]
That should already be supported, you just need to open the URL with Valent. For example:

```sh
flatpak run ca.andyholmes.Valent https://valent.andyholmes.ca
```

With files this is a bit easier, because you can use Files/Nautilus and use the "Open With..." to select Valent. So I guess the thing to do would be to mock up a UX/UI for sending URLs to devices, which I would happily enjoy discussing :)

**Reply by xusiai** [reply]
> That should already be supported, you just need to open the URL with Valent.

Good to know! Having the terminal option is sufficient for me. However a GUI options is perhaps worth a consideration for visibility, as send-url is a useful feature but as of now it is kind of hidden. 

Looking at the readme Valent seem to have support for sharing clipboard as well. Is this also doable from desktop to phone?

**Reply by andyholmes** [reply]
> However a GUI options is perhaps worth a consideration for visibility, as send-url is a useful feature but as of now it is kind of hidden.

Yeah, I started prototyping one :)

> Looking at the readme Valent seem to have support for sharing clipboard as well. Is this also doable from desktop to phone?

Yes, that should be automatic. I think you need to go into the device preferences and turn it on for each one though, since it's sort of a privacy concern.

![image](https://github.com/andyholmes/valent/assets/1265208/89adf5a5-3024-48fc-a0ac-c2121dc93b81)

**Reply by xusiai** [reply]
> Yes, that should be automatic. I think you need to go into the device preferences and turn it on for each one though, since it's sort of a privacy concern.

Ah, then it is perhaps a limitation on my end then as I am not a Gnome user.

Thanks for developing Valent. It is a lot more pleasant to use than the KDE application when you're not running the KDE desktop environment.

**Reply by andyholmes** [reply]
> Ah, then it is perhaps a limitation on my end then as I am not a Gnome user.

Could be. There are two built-in plugins, one that uses a D-Bus service from Mutter and a fallback based on GTK's clipboard. The GTK clipboard will only work well on X11 though, so other Wayland environments will need to write/contribute a plugin.

> Thanks for developing Valent. It is a lot more pleasant to use than the KDE application when you're not running the KDE desktop environment.

Thanks :)

**Reply by andyholmes** [reply]
Here's some screenshots/screencasts of my attempts at a new dialog. Let me know what you think.

This where you would start when using "Open With..." or `flatpak run ca.andyholmes.Valent file:///text.txt`: 
![Screenshot from 2023-09-29 23-05-53](https://github.com/andyholmes/valent/assets/1265208/69a2fc07-a63d-4b4e-927d-8e348156e97c)

Here is the first page of that window, showing entering a URL:

[Screencast from 2023-09-29 23-17-43.webm](https://github.com/andyholmes/valent/assets/1265208/ae1c1b53-e716-41f4-9d48-176e85b0b788)

**Comment by xusiai** [comment]
Looks perfect!

**Reply by andyholmes** [reply]
Thanks for waiting so patiently! I've finally got the pull request lined up and ready to merge :)

**Comment by aayushrautela** [comment]
Possibilty to use nautilus-extensions to add a direct sendto entry??

**Comment by andyholmes** [comment]
I guess the `closes` keyword didn't close this, so I'll close it up now :)
